indra/llcommon/llrand.cpp

```diff
@@ -82,15 +82,15 @@ static LLSeedRand sRandomSeeder;
 inline F64 ll_internal_random_double()
 {
 #if LL_WINDOWS
-	return (F64)rand() / (F64)RAND_MAX; 
+	return (F64)rand() / (F64)(RAND_MAX+1);
 #else
 	return drand48();
 #endif
 }
 inline F32 ll_internal_random_float()
 {
 #if LL_WINDOWS
-	return (F32)rand() / (F32)RAND_MAX; 
+	return (F32)rand() / (F32)(RAND_MAX+1);
 #else
 	return (F32)drand48();
 #endif
```

Comment:
Couldn't the whole body of `ll_internal_random_float()` be:
```
inline F32 ll_internal_random_float()
{
    return F32(ll_internal_random_double());
}
```

Reply:
oh boy do I have some simplifications in mind for all this.  I should probably write a jira